packages/react-ui/src/components/Modal.tsx

```diff
@@ -0,0 +1,248 @@
+    maxWidth: "400px",
+    width: "100%",
+    boxShadow: "0 8px 32px rgba(0, 0, 0, 0.3)",
+    position: "relative" as const,
+  };
+
+  const headerStyle: CSSProperties = {
+    position: "relative",
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    marginBottom: "24px",
+  };
+
+  const titleStyle: CSSProperties = {
+    margin: 0,
+    fontFamily: '"SF Pro Text", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, sans-serif',
+    fontSize: "14px",
+    fontStyle: "normal",
+    fontWeight: "400",
+    lineHeight: "17px",
+    letterSpacing: "-0.14px",
+    color: theme.secondary,
+    fontFeatureSettings: '"liga" off, "clig" off',
+    textAlign: "center" as const,
+  };
+
+  const closeButtonStyle: CSSProperties = {
+    position: "absolute" as const,
+    right: 0,
+    top: "50%",
+    transform: "translateY(-50%)",
+    background: "none",
+    border: "none",
+    color: theme.text,
+    fontSize: "24px",
+    cursor: "pointer",
+    padding: "4px 8px",
+    lineHeight: 1,
+    transition: "color 0.2s",
+  };
+
+  const appIconStyle: CSSProperties = appIcon
+    ? {
+        width: "56px",
+        height: "56px",
+        borderRadius: "50%",
+        display: "block",
+        margin: "0 auto 24px",
+        objectFit: "cover" as const,
+      }
+    : {};
+
+  const buttonContainerStyle: CSSProperties = {
+    display: "flex",
+    flexDirection: "column" as const,
+    gap: "12px",
+  };
+
+  const dividerStyle: CSSProperties = {
+    display: "flex",
+    alignItems: "center",
+    margin: "24px 0",
+    color: theme.secondary,
+    fontSize: "14px",
+    textTransform: "uppercase" as const,
+  };
+
+  const dividerLineStyle: CSSProperties = {
+    flex: 1,
+    height: "1px",
+    backgroundColor: theme.secondary,
+  };
+
+  const dividerTextStyle: CSSProperties = {
+    padding: "0 12px",
+  };
+
+  const footerStyle: CSSProperties = {
+    marginTop: "24px",
+    textAlign: "center" as const,
+    color: theme.secondary,
+    fontSize: "12px",
+  };
+
+  const errorStyle: CSSProperties = {
+    backgroundColor: "rgba(220, 53, 69, 0.1)",
+    color: "#ff6b6b",
+    border: "1px solid rgba(220, 53, 69, 0.3)",
+    borderRadius: "8px",
+    padding: "12px",
+    marginBottom: "12px",
+    fontSize: "14px",
+  };
+
+  return (
+    <div style={overlayStyle} onClick={onClose}>
+      <div style={modalStyle} onClick={e => e.stopPropagation()}>
+        {/* Header */}
+        <div style={headerStyle}>
+          <h3 style={titleStyle}>Login or Sign Up</h3>
+          <button
+            style={closeButtonStyle}
+            onClick={onClose}
+            onMouseEnter={e => (e.currentTarget.style.color = theme.secondary)}
+            onMouseLeave={e => (e.currentTarget.style.color = theme.text)}
+          >
+            ×
+          </button>
+        </div>
+
+        {/* App Icon */}
+        {appIcon && <img src={appIcon} alt={appName || "App"} style={appIconStyle} />}
```

Comment:
Nit: it's hard to miss the `appName` default. I suggest we move it to prop destructuring.